File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
    let answer = 0;
    
    if (n % 2 === 1) { // 홀수
        for (let i = 1; i <= n; i += 2) {
            answer += i;
        }
    } else { // 짝수
        for (let i = 2; i <= n; i += 2) {
            answer += i * i;
        }
    }
    
    return answer;
}

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.kt
class Solution {
    fun solution(n: Int): Int {
        var answer = 0

        if (n % 2 == 1) { // 홀수일 때
            for (i in 1..n step 2) {
                answer += i
            }
        } else { // 짝수일 때
            for (i in 2..n step 2) {
                answer += i * i
            }
        }

        return answer
    }
}
